src/lib.rs

```diff
@@ -66,6 +66,10 @@ impl Pidlock {
         }
     }
 
+    pub fn state(&self) -> &PidlockState {
+        &self.state
```

Comment:
Is there a reason why someone might like to know the new or released states? Is it enough to know that it’s locked or not?

Reply:
yeah, I'm currently trying to use this crate in a scenario where I need to start a server if it is not running, which is implemented in the same binary.

My approach is to check the lock and if it is aquired, the server is running and I do not start it again. Unfortunately I misunderstood the "state" here and it is not working as expected. I guess I'll simply check whether the pidfile exists or not.

Reply:
What if the function worked more like this situation:

  pub fn is_locked(&self) -> bool {}

That way, the implementation details on how to figure out lock state is still hidden, but the end goal is the same. Right?

Reply:
Yep, sounds good.